Bottom: 4bb671154dedb3939d353bcdc278816b8669819c
Top:    7bc50d871007bc1015cf0e2b7f29b69d80d633e1
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-02-11 16:23:56 +0800

Refresh of debug-1

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index db3205e..b36482a 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -2013,28 +2013,15 @@ static int get_indirect(struct vhost_virtqueue *vq,
 static bool desc_is_avail(struct vhost_virtqueue *vq,
 			  struct vring_desc_packed *desc)
 {
-	if (vq->used_wrap_counter) {
-		if ((desc->flags & VRING_DESC_F_AVAIL) &&
-		   !(desc->flags & VRING_DESC_F_USED))
-			return true;
-	} else {
-		if (!(desc->flags & VRING_DESC_F_AVAIL) &&
-		     (desc->flags & VRING_DESC_F_USED))
-			return true;
-	}
-	return false;
-
-#if 0
-	return ((desc->flags & cpu_to_vhost16(vq, VRING_DESC_F_AVAIL)) ^
-		(desc->flags & cpu_to_vhost16(vq, VRING_DESC_F_USED)));
-#endif
+	return ((desc->flags & cpu_to_vhost16(vq, 1 << VRING_DESC_F_AVAIL)) ^
+		(desc->flags & cpu_to_vhost16(vq, 1 << VRING_DESC_F_USED)));
 }
 
 static void set_desc_used(struct vhost_virtqueue *vq,
 			  struct vring_desc_packed *desc, bool wrap_counter)
 {
-	__virtio16 flags = vhost16_to_cpu(vq, VRING_DESC_F_USED |
-					      VRING_DESC_F_AVAIL);
+	__virtio16 flags = vhost16_to_cpu(vq, 1 << VRING_DESC_F_USED |
+					      1 << VRING_DESC_F_AVAIL);
 
 	if (wrap_counter)
 		desc->flags = flags;
